src/directives/if.js

```diff
@@ -88,9 +88,11 @@ module.exports = {
     function contains (c) {
       var cur = start
       var next
+      var isElementNode
       while (next !== end) {
         next = cur.nextSibling
-        if (cur.contains(c.$el)) {
+        isElementNode = cur.nodeType === Node.ELEMENT_NODE
+        if (isElementNode && cur.contains(c.$el)) {
```

Comment:
Thanks for the fix! The situation can be a bit more complicated though, in the case where the component is a block instance (thus its `$el` can be a TextNode too).

So I think it's better to change the check to:

``` js
if (
  cur === c.$el || // both are textNodes
  (cur.contains && cur.contains(c.$el)) // other situations
) {
  return true
}
```